src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient_private.cs

```diff
@@ -106,11 +109,40 @@ Uri BuildUriForLocksRoute(string key, string label)
             return builder.Uri;
         }
 
-        void BuildBatchQuery(UriBuilder builder, SettingSelector selector)
+        private string EscapeReservedCharacters(string input)
```

Comment:
All this escaping logic should be added to Azure.Base UriExtensions. All our components will need to properly escape. 
cc: @pakrym 

Also, maybe we need to create a solid UriBuilder-like type in Azure.Base, instead of just an extension type.

Reply:
I'll try to prototype something.

Reply:
Created issue https://github.com/Azure/azure-sdk-for-net/issues/5647 to address that.